tests/integration/workflows/test_workflows_tasks_upload.py

```diff
@@ -173,3 +178,33 @@ def test_is_stale_data_returns_false_if_is_update_is_falsy(workflow_app):
     TestRecordMetadata.create_from_kwargs(index=False, has_pid=False)
     obj = workflow_object_class.create({})
     assert is_stale_data(obj, None) is False
+
+
+def test_regression_store_record_does_not_commit_when_error(workflow_app):
+    data = {
+        '$schema': 'http://localhost:5000/schemas/records/hep.json',
+        '_collections': ['Literature'],
+        'document_type': ['article'],
+        'titles': [{'title': 'title'}],
+    }
+    eng = MagicMock(workflow_definition=MagicMock(data_type='hep'))
+
+    obj = workflow_object_class.create(data)
+
+    def raise_func(*args, **kwargs):
+        raise Exception()
+
+    record_count = RecordMetadata.query.count()
+    assert record_count == 0
+
+    with patch.object(
+        InspireRecord,
+        'download_documents_and_figures',
+        raise_func
+    ):
+        # pytest.raises catches the exception and makes the test passing immediately
```

Comment:
not sure I understand this comment: do you mean
```python
with pytest.raises(Exception):
    store_record(obj, eng)
record_count = RecordMetadata.query.count()
assert  record_count == 0
```
doesn't work? (doesn't really matter though, this way of doing it is fine, as you're not testing that it raises, but that the db state is correct).

Reply:
Exactly. I wrote this test (using `pytest.raises`) before writing the bugfix, expecting it to fail but it was passing, since the exception was caught properly. This way it failed properly.